go/worker/common/committee/group.go

```diff
@@ -527,8 +531,13 @@ func (g *Group) HandlePeerMessage(unusedPeerID signature.PublicKey, message *p2p
 		// Ensure that both peers have the same view of the current group. If this
 		// is not the case, this means that one of the nodes processed an epoch
 		// transition and the other one didn't.
-		if message.GroupVersion != g.activeEpoch.groupVersion {
-			return nil, fmt.Errorf("group version mismatch")
+		switch {
+		case message.GroupVersion < g.activeEpoch.groupVersion:
+			// Stale messages will never become valid.
+			return nil, p2pError.Permanent(fmt.Errorf("group version in the past"))
+		case message.GroupVersion > g.activeEpoch.groupVersion:
+			// Messages from the future may eventually become valid.
+			return nil, p2pError.Permanent(fmt.Errorf("group version from the future"))
```

Comment:
The comment does not match the code, this should probably not be a permanent error?

Reply:
Whoops, copy/paste issue.